cumulusci/utils/xml/metadata_tree.py

```diff
@@ -0,0 +1,251 @@
+        element.text = text
+        return self._wrap_element(element)
+
+    def __getattr__(self, childname):
+        return self._get_child(childname)
+
+    def __getitem__(self, item: Union[int, str]):
+        if isinstance(item, int):
+            children = self._parent.findall(self._element.tag)
+            return self._wrap_element(children[item])
+        elif isinstance(item, str):
+            return self._get_child(item)
+        else:
+            raise TypeError(type(item))
+
+    def append(self, tag: str, text: str = None):
+        '''Append a new element at the appropriate place.
+
+        If the parent element (self) already one or more children that match,
+        the new element follows the last one.
+
+        Otherwise, the new element goes to the bottom of the parent (self).
+
+        >>> Package = metadata_tree.fromstring(b"""<?xml version="1.0" encoding="UTF-8"?>
+        ... <Package xmlns="http://soap.sforce.com/2006/04/metadata">
+        ...     <types>
+        ...         <members>CaseReason</members>
+        ...         <members>CaseStatus</members>
+        ...         <members>CaseType</members>
+        ...         <name>StandardValueSet</name>
+        ...     </types>
+        ...     <version>46.0</version>
+        ... </Package>""")
+        >>> Package.types.append("members", "CaseOfBeer")
+        <members>CaseOfBeer</members> element
+        >>> print(Package.types.tostring())
+        <types xmlns="http://soap.sforce.com/2006/04/metadata">
+            <members>CaseReason</members>
+            <members>CaseStatus</members>
+            <members>CaseType</members>
+            <members>CaseOfBeer</members>
+            <name>StandardValueSet</name>
+        </types>
+        '''
+        newchild = self._create_child(tag, text)
+        same_elements = self._element.findall(self._add_namespace(tag))
+        if same_elements:
+            last = same_elements[-1]
+            index = self._element.index(last)
+            self._element.insert(index + 1, newchild._element)
+        else:
+            self._element.append(newchild._element)
+        return newchild
+
+    def insert(self, index: int, tag: str, text: str = None):
+        """Insert at a particular index.
+
+        Tag and text can be supplied. Return value is the new element.
+
+        append is preferable because it ensures that nodes are inserted
+        in the right "group".
+
+        If you need to get to a particular place in a "group" then insertBefore
+        and insertAfter are preferable.
+
+        If all else fails then you can use this one to precisely insert right
+        where you want it but you're responsible for adhering to Salesforce's
+        grouping rules.
+        """
+        newchild = self._create_child(tag, text)
+        self._element.insert(index, newchild._element)
+        return newchild
+
+    def insertBefore(self, oldElement: "MetadataElement", tag: str, text: str = None):
```

Comment:
Is there a reason for lowerCamel rather than snake_case?

Reply:
I guess it was a subconscious reference to this:

https://www.w3schools.com/jsref/met_node_insertbefore.asp

I don't feel strongly about it though. I could go either way.

Reply:
Let's use camel case. This is Python, not Javascript.

Reply:
9eae9fee

Although I will note that insertBefore shows up twice in the stdlib.